server/accounts.go

```diff
@@ -2651,7 +2693,292 @@ func (ur *URLAccResolver) Fetch(name string) (string, error) {
+			respondToUpdate(s, resp, pubKey, "jwt update resulted in error", err)
+		} else {
+			respondToUpdate(s, resp, pubKey, "jwt updated", nil)
+		}
+	}); err != nil {
+		s.Errorf("Error setting up update handling: %v", err)
+	} else if subLookupReq, err = s.sysSubscribe(accountLookupRequest, func(_ *subscription, _ *client, subj, reply string, msg []byte) {
+		// respond to lookups with our version
+		if reply == "" {
+			return
+		}
+		tk := strings.Split(subj, tsep)
+		if len(tk) != accUpdateTokens {
+			return
+		}
+		if theJWT, err := dr.DirJWTStore.LoadAcc(tk[accUpdateAccIndex]); err != nil {
+			s.Errorf("Merging resulted in error: %v", err)
+		} else {
+			s.sendInternalMsgLocked(reply, "", nil, []byte(theJWT))
+		}
+	}); err != nil {
+		s.Errorf("Error setting up pack response handling: %v", err)
+	} else if subPackReq, err = s.sysSubscribeQ(accountPackRequest, "responder",
+		// respond to pack requests with one or more pack messages
+		// an empty message signifies the end of the response responder
+		func(_ *subscription, _ *client, _, reply string, theirHash []byte) {
+			if reply == "" {
+				return
+			}
+			ourHash := dr.DirJWTStore.Hash()
+			if bytes.Equal(theirHash, ourHash[:]) {
+				s.sendInternalMsgLocked(reply, "", nil, []byte{})
+				s.Debugf("pack request matches hash %x", ourHash[:])
+			} else if err := dr.DirJWTStore.PackWalk(1, func(partialPackMsg string) {
+				s.sendInternalMsgLocked(reply, "", nil, []byte(partialPackMsg))
+			}); err != nil {
+				// let them timeout
+				s.Errorf("pack request error: %v", err)
+			} else {
+				s.Debugf("pack request hash %x - finished responding with hash %x")
+				s.sendInternalMsgLocked(reply, "", nil, []byte{})
+			}
+		}); err != nil {
+		s.Errorf("Error setting up pack request handling: %v", err)
+	} else if subPackResp, err = s.sysSubscribe(packRespIb, func(_ *subscription, _ *client, _, _ string, msg []byte) {
+		// embed pack responses into store
+		hash := dr.DirJWTStore.Hash()
+		if len(msg) == 0 { // end of response stream
+			s.Debugf("Merging Finished and resulting in: %x", dr.DirJWTStore.Hash())
+			return
+		} else if err := dr.DirJWTStore.Merge(string(msg)); err != nil {
+			s.Errorf("Merging resulted in error: %v", err)
+		} else {
+			s.Debugf("Merging succeeded and changed %x to %x", hash, dr.DirJWTStore.Hash())
+		}
+	}); err != nil {
+		s.Errorf("Error setting up pack response handling: %v", err)
+	} else {
+		// periodically send out pack message
+		go func() {
+			ticker := time.NewTicker(dr.syncInterval)
+			for {
+				select {
+				case <-quit:
+					ticker.Stop()
+					return
+				case <-ticker.C:
+				}
+				ourHash := dr.DirJWTStore.Hash()
+				s.Debugf("Checking store state: %x", ourHash)
+				s.sendInternalMsgLocked(accountPackRequest, packRespIb, nil, ourHash[:])
+			}
+		}()
+		s.Noticef("Managing all jwt in exclusive directory %s", dr.directory)
+	}
+	closer := func() {
+		close(quit)
+		s.sysUnsubscribe(subUpdate)
```

Comment:
We are not unsubscribing any other internal subs when shutting down eventing, so wonder if that is a problem or if this is superfluous here. If the later, it would simplify a little bit code. You would simply have to select on `s.quitCh` instead of `quit` for the above go routine to end.

Reply:
will give this a try

Reply:
done